TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+    dispatch_once( &predicate, ^{
+        sharedInstance = [[ self alloc ] init];
+    });
+    
+    return sharedInstance;
+}
+
+#pragma mark - Core Data procedures
+
+-(NSNumber *)getNewId{
+    
+    NSDate *from1970 = [[NSDate alloc] initWithTimeIntervalSinceNow:0];
+    NSNumber *newid = [NSNumber numberWithUnsignedInteger:[[NSNumber numberWithUnsignedInteger:[from1970 timeIntervalSince1970]] unsignedIntegerValue]];
+
+    return newid;
+}
+
+
+-(void)addTask:(NSMutableDictionary *)dictionaryTask taskId:(NSNumber *)lidNumber{
+    
+    Tasks *task;
+    
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Tasks" inManagedObjectContext:self.managedObjectContext];
+
+    NSPredicate *predicateLid = [NSPredicate predicateWithFormat:@"lid = %@", [dictionaryTask valueForKey:@"lid"]];
+
+    NSPredicate *compPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:[NSArray arrayWithObjects:predicateLid, nil]];
+    [fetchRequest setPredicate:compPredicate];
+
+    NSError *requestError = nil;
+    
+    [fetchRequest setIncludesPendingChanges:YES];
+    [fetchRequest setEntity:entity];
+    
+    NSArray *items = [self.managedObjectContext executeFetchRequest:fetchRequest error:&requestError];
+    
+    NSLog(@"records in Tasks: %ld", (long) [items count]);
+
+    if ([items count] > 0){
+        task = [items objectAtIndex:0];
+    }else{
+        task = [NSEntityDescription insertNewObjectForEntityForName:@"Tasks" inManagedObjectContext:self.managedObjectContext];
+    }
+    
+    [task setValuesForKeysWithDictionary:[NSDictionary dictionaryWithDictionary:dictionaryTask]];
+    
+    NSError *savingError;
+    
+    @try {
+        
+        if ([self.managedObjectContext save:&savingError]){
+            NSLog(@"Tasks saved Ok");
+        }else{
+            NSLog(@"error %@", savingError);
+        }
+        
+    } @catch (NSException *exception) {
+        
+        NSLog(@"exception: %@", [exception debugDescription]);
+        
+    }
+
+}
+
+
+-(NSArray *)getTasks{
+    
+    NSArray *resultArray;
+    NSMutableArray *fetchedTasks = [[NSMutableArray alloc] init];
+    Tasks *task;
+    
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Tasks" inManagedObjectContext:self.managedObjectContext];
+    
+    NSNumber *status = [NSNumber numberWithUnsignedInteger:3];
+
+    NSPredicate *predicateLid = [NSPredicate predicateWithFormat:@"status < %@", status];
+    
+    NSPredicate *compPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:[NSArray arrayWithObjects:predicateLid, nil]];
+    [fetchRequest setPredicate:compPredicate];
+    
+    NSInteger sortOrder = [[NSUserDefaults standardUserDefaults] integerForKey:@"sortOrder"];
+
+    NSSortDescriptor *sortingPredicate;
+
+    
+    switch (sortOrder) {
+        case 0:{
```

Comment:
These should be named constants.

Reply:
it worse! he used position from this [array](https://github.com/farcaller/doitsofware.test.assignment/pull/1/files/e78934da3f1caba3f030c026f7c1dbf51d08fe05#diff-333c31d6db1f24ba46a30e610e484a93R39) as values for this enum